src/drivers/intel/ssp/ssp.c

```diff
@@ -410,21 +418,23 @@ static int ssp_set_config_tplg(struct dai *dai, struct ipc_config_dai *common_co
 	sscr2 |= (ssp->params.quirks & SOF_DAI_INTEL_SSP_QUIRK_PSPSRWFDFD) ?
 		SSCR2_PSPSRWFDFD : 0;
 
-	if (!config->ssp.mclk_rate ||
-	    config->ssp.mclk_rate > ssp_freq[MAX_SSP_FREQ_INDEX].freq) {
-		dai_err(dai, "ssp_set_config(): invalid MCLK = %d Hz (valid < %d)",
-			config->ssp.mclk_rate,
-			ssp_freq[MAX_SSP_FREQ_INDEX].freq);
-		ret = -EINVAL;
-		goto out;
-	}
+	/* not every codec reqire mclk */
+	if (config->ssp.mclk_rate) {
```

Comment:
config->ssp.mclk_rate
if it is zero, previous logic will goto out, current does not, is this expected?

Reply:
Yes, in previous logic, you need to specify a mclk id and rate even the codec does not need it. Here we use the zero to specify this situation.